src/bin/drone.rs

```diff
@@ -87,6 +90,30 @@ fn main() {
         request_cap = None;
     }
 
+    // Set up gossip functionality
+    let exit = Arc::new(AtomicBool::new(false));
+    let testnode = TestNode::new_localhost();
+    let extra_data = testnode.data.clone();
+    let crdt = Arc::new(RwLock::new(Crdt::new(extra_data).expect("Crdt::new")));
+    let window = Arc::new(RwLock::new(vec![]));
+    let ncp = Ncp::new(
+        &crdt.clone(),
+        window,
+        None,
+        testnode.sockets.gossip,
+        testnode.sockets.gossip_send,
+        exit.clone(),
+    ).unwrap();
+    let leader_entry_point = NodeInfo::new_entry_point(leader.contact_info.ncp);
+    crdt.write().unwrap().insert(&leader_entry_point);
+
+    // Block until leader's correct contact info is received
+    while crdt.read().unwrap().leader_data().is_none() {}
```

Comment:
this needs a sleep

Reply:
Sure. Between the crdt write and read? How long and why?

Reply:
this will spin the thread without yielding to the rest of the system to actually process anything to  get the leader_data.  if you sleep, that tells the OS that it can do other useful things.  100ms is probably good enough.  

maybe pull this out into a function that polls for X seconds and returns an error if it fails.  otherwise we may end up with this stuck in a misconfigured test.

Reply:
The recent push implements a configurable timeout for both wallet and drone gossip-poll. (Timeout can also be none, if blocking is desired.) I plugged in some starting values in the bash scripts that are reasonable for my local machine. @aeyakovenko , do you have a feel for how long is reasonable for tests/automation? Maybe a question for @mvines ?
Presumably the wallet timeout can be short, since we expect a full node to be up, but I'm not clear on how long we would expect between drone and leader boot.